morpheus/_lib/src/python_modules/messages.cpp

```diff
@@ -110,10 +110,17 @@ PYBIND11_MODULE(messages, m)
     mrc::node::EdgeConnector<std::shared_ptr<morpheus::MultiResponseProbsMessage>,
                              std::shared_ptr<morpheus::MultiMessage>>::register_converter();
 
+    py::class_<MutableCtxMgr, std::shared_ptr<MutableCtxMgr>>(m, "MutableCtxMgr")
+        .def("__enter__", &MutableCtxMgr::enter, py::return_value_policy::reference)
+        .def("__exit__", &MutableCtxMgr::exit)
+        .def_property_readonly("df", &MutableCtxMgr::df_property, py::return_value_policy::reference);
+
     py::class_<MessageMeta, std::shared_ptr<MessageMeta>>(m, "MessageMeta")
         .def(py::init<>(&MessageMetaInterfaceProxy::init_python), py::arg("df"))
         .def_property_readonly("count", &MessageMetaInterfaceProxy::count)
-        .def_property_readonly("df", &MessageMetaInterfaceProxy::get_data_frame, py::return_value_policy::move)
+        .def_property_readonly("df", &MessageMetaInterfaceProxy::df_property, py::return_value_policy::move)
```

Comment:
Can we show a deprecation message when this is used? Here is an example in the MRC repo: https://github.com/nv-morpheus/MRC/pull/261/files#diff-443d85d7ff4910fdc3e77cfc71629f85257ff6f21ae0d4f23733504cbf904678R63

Reply:
We do have a deprecation warning in `meta.cpp`:
```
pybind11::object MessageMetaInterfaceProxy::df_property(MessageMeta& self)
{
    PyErr_WarnEx(
        PyExc_DeprecationWarning,
        "Warning the df property returns a copy, please use the copy_dataframe method or the mutable_dataframe "
        "context manager to modify the DataFrame in-place instead.",
        1);

    return MessageMetaInterfaceProxy::get_data_frame(self);
}
```